init.d/bootmisc.in

```diff
@@ -25,19 +25,16 @@ cleanup_tmp_dir()
 	local dir="$1"
 
 	if ! [ -d "$dir" ]; then
-		mkdir -p "$dir" || return $?
+		mkdir -p "$dir"
+		return $?
 	fi
 	checkpath -W "$dir" || return 1
 	chmod a+rwt "$dir" 2> /dev/null
 	cd "$dir" || return 1
 	if yesno $wipe_tmp; then
 		ebegin "Wiping $dir directory"
 
-		# Faster than raw find
-		if ! rm -rf -- [!ajlq\.]* 2>/dev/null ; then
-			# Blah, too many files
-			find . -maxdepth 1 -name '[!ajlq\.]*' -exec rm -rf -- {} +
-		fi
+		find "${dir}" -xdev -maxdepth 1 -name '[!ajlq\.]*' -exec rm -rf -- {} +
```

Comment:
this doesn't actually ignore submounts because `rm -rf` is still being invoked on the topdir.  e.g. /tmp/foo/bar mountpoints.